core/src/dird/ua_run.cc

```diff
@@ -1792,14 +1816,27 @@ static bool ScanCommandLineArguments(UaContext* ua, RunContext& rc)
             rc.job_name = ua->argv[i];
             kw_ok = true;
             break;
-          case 1: /* JobId */
+          case 1: /* JobId */ {
             if (rc.jid && !rc.mod) {
               ua->SendMsg(_("JobId specified twice.\n"));
               return false;
             }
+            JobControlRecord* jcr = ua->jcr;
             rc.jid = ua->argv[i];
+            std::string jobIdInput = ua->argv[i];
+            std::vector jobIdList = split_string(jobIdInput, ',');
+            for (auto& jobId : jobIdList) {
+              bool foundJob;
+              foundJob = jcr->db->FindJobById(jcr, jobId);
+              if (!foundJob) {
+                ua->SendMsg("JobId not found in catalog. \n");
+                return false;
+              }
+            }
+
             kw_ok = true;
             break;
+          }
```

Comment:
```suggestion
          case 1: /* JobId */ {
            if (rc.jid && !rc.mod) {
              ua->SendMsg(_("JobId specified twice.\n"));
              return false;
            }
            rc.jid = ua->argv[i];
            if (!ua->db->FindJobById(ua->jcr, rc.jid)) {
              ua->SendMsg("JobId not found in catalog. \n");
              return false;
            }

            kw_ok = true;
            break;
```
Since at the moment we do not handle comma separated jobid's here, there is no need to do it in this check alone. Even if all of them are correct, the proceeding jobs would only take the first jobid in consideration.

Reply:
turns out that it is just the whole string that used to get passed, and whoever receives it, does whatever he wants with it. Some functions do split them, others don't.